data/crac-io/crac-io-json/src/main/java/com/farao_community/farao/data/crac_io_json/deserializers/CounterTradeRangeActionArrayDeserializer.java

```diff
@@ -0,0 +1,118 @@
+package com.farao_community.farao.data.crac_io_json.deserializers;
+
+import com.farao_community.farao.commons.FaraoException;
+import com.farao_community.farao.data.crac_api.Crac;
+import com.farao_community.farao.data.crac_api.range_action.CounterTradeRangeActionAdder;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.powsybl.iidm.network.Country;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static com.farao_community.farao.data.crac_io_json.JsonSerializationConstants.*;
+
+/**
+ * @author Gabriel Plante {@literal <gabriel.plante_externe at rte-france.com>}
+ */
+public final class CounterTradeRangeActionArrayDeserializer {
+    private CounterTradeRangeActionArrayDeserializer() {
+    }
+
+    public static void deserialize(JsonParser jsonParser, String version, Crac crac, Map<String, String> networkElementsNamesPerId) throws IOException {
+        if (networkElementsNamesPerId == null) {
+            throw new FaraoException(String.format("Cannot deserialize %s before %s", INJECTION_RANGE_ACTIONS, NETWORK_ELEMENTS_NAME_PER_ID));
+        }
+        while (jsonParser.nextToken() != JsonToken.END_ARRAY) {
+            CounterTradeRangeActionAdder counterTradeRangeActionAdder = crac.newCounterTradeRangeAction();
+
+            while (!jsonParser.nextToken().isStructEnd()) {
+                addElement(counterTradeRangeActionAdder, jsonParser, version);
+            }
+            if (getPrimaryVersionNumber(version) <= 1 && getSubVersionNumber(version) < 3) {
+                // initial setpoint was not exported then, set default value to 0 to avoid errors
+                counterTradeRangeActionAdder.withInitialSetpoint(0);
+            }
+            counterTradeRangeActionAdder.add();
+        }
+    }
+
+    private static void addElement(CounterTradeRangeActionAdder counterTradeRangeActionAdder, JsonParser jsonParser, String version) throws IOException {
```

Comment:
Code duplication (Injection|HVDC|PST)RangeActionArrayDeserializer, Could this be avoided ?

Reply:
You can create a common class also, with a "addElement" method that handles common attribues. This method would return a boolean to indicate if it could handle this attribute (should be feasible at least for adders that extend StandardRangeActionAdder, ie HVDC, injection and CT). If not, then the specific deserializer should try to import the attribute.